ruff_sync.py

```diff
@@ -240,6 +260,10 @@ async def sync(
 def main() -> None:
     args = PARSER.parse_args()
     # config = get_config(args.source)
+
+    # Convert non-raw github upstream url to the raw equivalent
+    args.upstream = github_url_to_raw_url(args.upstream)
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Validate the `args.upstream` URL format before conversion.

Before converting `args.upstream` with `github_url_to_raw_url`, ensure it's in a valid URL format to avoid runtime errors.

```suggestion
import re

url_pattern = re.compile(r'https?://github\.com/[\w-]+/[\w-]+/?')
if url_pattern.match(args.upstream):
    args.upstream = github_url_to_raw_url(args.upstream)
```